.github/workflows/build-test-distribute.yml

```
@@ -14,111 +14,146 @@ on:
     - cron: '0 17 * * *'  # Run every day at 17:00 UTC
     #- cron: '0 6 * * *'  # Run every day at 6 AM UTC
 
-#UPLOAD_ARTIFACTS condition: ${{ github.event_name == 'push' || contains(github.event.pull_request.labels.*.name, 'full-ci') }}
-
+jobs:
+  # NOTE: all output values are strings
+  # use the ${{ needs.config.outputs.bool-param == 'true' }} construction to check boolean values
+  config:
+    uses: ./.github/workflows/config.yml
 
-env:
-  VCPKG-VERSION: '2024.10.21'
+  debug-config:
+    needs: config
+    runs-on: ubuntu-latest
+    steps:
+      - name: Print config
+        run: |
+          echo << EOF
+          ${{ toJSON( needs.config.outputs ) }}
+          EOF
 
-jobs:
   prepare-image:
```

sometimes we can skip this job
```suggestion
  prepare-image:
  if: ${{ needs.config.outputs.need_linux_image_rebuild == 'true'  || needs.config.outputs.need_windows_vcpkg_rebuild == 'true' }}
```